pycbc/inference/models/hierarchical.py

```diff
@@ -599,3 +604,349 @@ def _loglikelihood(self):
+    for ground-based detectors, please use `RelativeTimeDom`.
+
+    Although this likelihood model is used for multiband parameter
+    estimation, users can still use it for other purposes, such as
+    GW + EM parameter estimation, in this case, please use `RelativeTimeDom`
+    for the GW data, for the likelihood of EM data, there is no restrictions.
+    """
+    name = 'joint_primary_marginalized'
+
+    def __init__(self, variable_params, submodels, **kwargs):
+        super().__init__(variable_params, submodels, **kwargs)
+
+        # assume the ground-based submodel as the primary model
+        self.primary_model = self.submodels[kwargs['primary_lbl'][0]]
+        self.primary_lbl = kwargs['primary_lbl'][0]
+        self.other_models = self.submodels.copy()
+        self.other_models.pop(kwargs['primary_lbl'][0])
+        self.other_models = list(self.other_models.values())
+
+    def write_metadata(self, fp, group=None):
+        """Adds metadata to the output files
+
+        Parameters
+        ----------
+        fp : pycbc.inference.io.BaseInferenceFile instance
+            The inference file to write to.
+        group : str, optional
+            If provided, the metadata will be written to the attrs specified
+            by group, i.e., to ``fp[group].attrs``. Otherwise, metadata is
+            written to the top-level attrs (``fp.attrs``).
+        """
+        super().write_metadata(fp, group=group)
+        sampattrs = fp.getattrs(group=fp.samples_group)
+        # if a group is specified, prepend the lognl names with it
+        if group is None or group == '/':
+            prefix = ''
+        else:
+            prefix = group.replace('/', '__')
+            if not prefix.endswith('__'):
+                prefix += '__'
+        try:
+            for lbl, model in self.submodels.items():
+                sampattrs['{}lognl'.format(prefix + '%s__' % lbl)
+                          ] = model.lognl
+        except AttributeError:
+            pass
+
+    def total_loglr(self):
+        r"""Computes the total log likelihood ratio,
+
+        .. math::
+
+            \log \mathcal{L}(\Theta) = \sum_i
+                \left<h_i(\Theta)|d_i\right> -
+                \frac{1}{2}\left<h_i(\Theta)|h_i(\Theta)\right>,
+
+        at the current parameter values :math:`\Theta`.
+
+        Returns
+        -------
+        float
+            The value of the log likelihood ratio.
+        """
+        # calculate <d-h|d-h> = <h|h> - 2<h|d> + <d|d> up to a constant
+
+        # note that for SOBHB signals, ground-based detectors dominant SNR
+        # and accuracy of (tc, ra, dec)
+        self.primary_model.return_sh_hh = True
+        sh_primary, hh_primary = self.primary_model.loglr
+        self.primary_model.return_sh_hh = False
+
+        margin_names = list(
+            self.primary_model.marginalize_vector_params.keys())
+        margin_names.remove('logw_partial')
```

Comment:
Do you need this removal? What happens if you don't do this?

Reply:
@ahnitz Those are related to updating parameters in the LISA model, so there is no need for `logw_partial`  in the LISA model.